pp_sys.c

```diff
@@ -4323,7 +4323,7 @@ PP(pp_system)
             sv_2mortal(copysv);
             if (SvPOK(origsv) || SvPOKp(origsv)) {
                 pv = SvPV_nomg(origsv, len);
-                sv_setpvn(copysv, pv, len);
+                sv_setpvn_fresh(copysv, pv, len);
```

Comment:
Won't this assert?  copysv is created as 
```
copysv = newSV_type(SVt_PVNV);
```

(but only on Win32)

Reply:
Yes, good spot. (Not hit by the existing tests?) It seems like for this case it would be fine to relax the assert condition, but I haven't had time to think over whether that's a wise choice overall.

Reply:
Neither of the Win32 CI builds is a -DDEBUGGING build, so the assertions wouldn't have failed.

Reply:
I've relaxed the type assertions in both the new functions. Can you foresee any problems with that?

(If you're happy, this PR still needs me to rebase and and do a squash.)

Reply:
Making it `>= SVt_PV` and maybe `<= SVt_PVMG` would be more general.

Reply:
Thanks, I've made that change and rebased. Is there anything else you've spotted before I squash?

Reply:
I don't see any further problems.